src/stcal/linearity/linearity.py

```diff
@@ -0,0 +1,202 @@
+import numpy as np
+
+
+def linearity_correction(data, gdq, pdq, lin_coeffs, lin_dq, dqflags):
+    """
+    Short Summary
+    -------------
```

Comment:
This is a bit of nitpicking and I know it comes from jwst  but let's try to make docstrings consistent.

Remove the section definition above and have a one sentence description, e.g. `Apply linearity correction to non-saturated pixels in individual groups.`

Then a blank line and  the paragraph follows.